ironic/resource_ironic_allocation_v1.go

```diff
@@ -0,0 +1,160 @@
+package ironic
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack/baremetal/v1/allocations"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+// Schema resource definition for an Ironic allocation.
+func resourceAllocationV1() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAllocationV1Create,
+		Read:   resourceAllocationV1Read,
+		Delete: resourceAllocationV1Delete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"resource_class": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"candidate_nodes": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional: true,
+				ForceNew: true,
+			},
+			"traits": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional: true,
+				ForceNew: true,
+			},
+			"extra": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				ForceNew: true,
+			},
+			"node_uuid": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"last_error": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+// Create an allocation, including driving Ironic's state machine
+func resourceAllocationV1Create(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*gophercloud.ServiceClient)
+
+	result, err := allocations.Create(client, allocationSchemaToCreateOpts(d)).Extract()
+	if err != nil {
+		return err
+	}
+
+	d.SetId(result.UUID)
+
+	// Wait for state to change from allocating
+	var state string
+	timeout := 1 * time.Minute
```

Comment:
Is this a reasonable time period to wait for an allocation's state to change from `allocating` to either `error` or `active`?

Reply:
More than. It's usually 1-3 seconds.

P.S.
I will check the whole patch tomorrow.